var/spack/repos/builtin/packages/libelf/package.py

```diff
@@ -29,8 +29,23 @@ class Libelf(AutotoolsPackage):
 
     provides("elf@0")
 
+    # configure: error: neither int nor long is 32-bit
+    depends_on("automake", when="platform=darwin target=aarch64:", type="build")
+    depends_on("autoconf", when="platform=darwin target=aarch64:", type="build")
+    depends_on("libtool", when="platform=darwin target=aarch64:", type="build")
+    depends_on("m4", when="platform=darwin target=aarch64:", type="build")
+
+    @property
+    def force_autoreconf(self):
+        return self.spec.satisfies("platform=darwin target=aarch64:")
```

Comment:
Homebrew and Conan use the same approach

Reply:
Yep, no problem